Source/ACE.Common/Cryptography/ISAAC.cs

```diff
@@ -1,11 +1,51 @@
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace ACE.Common.Cryptography
 {
     public class ISAAC
     {
-        public static byte[] ClientSeed { get; } = { 0x60, 0xAF, 0x54, 0x6D }; // C->S
-        public static byte[] ServerSeed { get; } = { 0xCD, 0xD7, 0xEB, 0x45 }; // S->C
+#if NETDIAG
+        private Queue<ISAAC> ancestry = new Queue<ISAAC>();
+        public ISAAC Copy()
+        {
+            ISAAC newCopy = new ISAAC(null);
+            newCopy.SetInternalState(offset, a, b, c, mm, randRsl, ancestry);
+            return newCopy;
+        }
+        public ISAAC Parent
+        {
+            get
+            {
+                return ancestry.ToArray().LastOrDefault();
+            }
+        }
+#endif
+        private void SetInternalState(uint offset, uint a, uint b, uint c, uint[] mm, uint[] randRsl)
+        {
+            this.offset = offset;
+            this.a = a;
+            this.b = b;
+            this.c = c;
+            this.mm = new uint[256];
+            this.randRsl = new uint[256];
+            Array.Copy(mm, this.mm, mm.Length);
+            Array.Copy(randRsl, this.randRsl, randRsl.Length);
+        }
+#if NETDIAG
+        private void SetInternalState(uint offset, uint a, uint b, uint c, uint[] mm, uint[] randRsl, Queue<ISAAC> ancestry)
+        {
+            SetInternalState(offset, a, b, c, mm, randRsl);
+            var g = ancestry.ToArray();
+            var f = new ISAAC[g.Length];
+            Array.Copy(g, f, g.Length);
+            this.ancestry = new Queue<ISAAC>(f);
+        }
+#endif
+        public static byte[] ClientSeed { get; } = { 0x60, 0xAF, 0x54, 0x6D }; // C->S    // TO-DO: needs to be random
+        public static byte[] ServerSeed { get; } = { 0xCD, 0xD7, 0xEB, 0x45 }; // S->C    // TO-DO: needs to be random
```

Comment:
Need to see if I can move Physics.Common.Random over to ACE.Common, to accommodate random initial ISSAC seeds. Want to get opinion from others first. Perhaps not simple without a performance hit so need to do it right, perhaps much search-replace is needed. I prefer not to make this change in this PR.